oem/ibm/libpldmresponder/file_io.cpp

```diff
@@ -745,12 +761,32 @@ Response Handler::writeFileByType(const pldm_msg* request, size_t payloadLength)
         return response;
     }
 
+    fileack_status_metadata metaDataObj;
     rc = handler->write(reinterpret_cast<const char*>(
                             request->payload + PLDM_RW_FILE_BY_TYPE_REQ_BYTES),
-                        offset, length, oemPlatformHandler);
+                        offset, length, oemPlatformHandler, metaDataObj);
+
     encode_rw_file_by_type_resp(request->hdr.instance_id,
                                 PLDM_WRITE_FILE_BY_TYPE, rc, length,
                                 responsePtr);
+
+    if (fileType == PLDM_FILE_TYPE_USER_PASSWORD_AUTHENTICATION ||
+        fileType == PLDM_FILE_TYPE_USER_PASSWORD_CHANGE)
+    {
+        sdbusplus::message::object_path path;
+        if (rc == PLDM_SUCCESS)
```

Comment:
instead of if else condition, optimize like if(rc!=PLDM_SUCCESS){ error() return;}

Reply:
I do not want to return from here and just print the trace if write failed